src/hooks/use-play-lesson.ts

```diff
@@ -1,46 +1,127 @@
+  const [firstCard, ...restCards] = getLessonCards(deck, numCards);
+  const [currentCard, setCurrentCard] = useState<LessonCard>(firstCard);
+  const [upcomingCards, setUpcomingCards] = useState(restCards);
+  const [completedCards, setCompletedCards] = useState<LessonCard[]>([]);
+  const [isPaused, setIsPaused] = useState(true);
+  const { pauseAudio, resumeAudio, playAudio, clearAudio, activeCardSide, activeCard } =
+    usePlayCardAudio();
+
+  return {
+    currentCard: currentCard?.card,
     activeCardSide,
-    playTermAndDefinition,
-    pause,
-    resume,
-    getTotalCardDuration,
-  } = usePlayCardAudio();
+    completedCards: [...completedCards],
+    play: play,
+    pause: pauseCard,
+    skip: skipCard,
+    replay: replayCard,
+  };
+
+  function play() {
+    setIsPaused(false);
+    playCard(currentCard, upcomingCards, completedCards);
+  }
 
-  return { activeCardKey, activeCardSide, startLesson, pauseLesson, resumeLesson };
+  function skipCard() {
+    clearAudio();
+    const next = nextCard(currentCard, upcomingCards, completedCards);
+    if (!isPaused) {
+      playCard(next.currentCard, next.upcomingCards, next.completedCards);
+    }
+  }
 
-  function startLesson(deck: Deck) {
-    lessonCards.current = buildLesson(deck);
-    playNextCard();
+  function replayCard() {
+    // TODO: will want a stopwatch to both replay current card and play previous card
+    clearAudio();
+    const next = previousCard(currentCard, upcomingCards, completedCards);
+    if (!isPaused) {
+      playCard(next.currentCard, next.upcomingCards, next.completedCards);
+    }
   }
 
-  async function playNextCard() {
-    if (lessonCards.current.length === 0) {
+  function pauseCard() {
+    setIsPaused(true);
+    pauseAudio();
+  }
+
+  async function playCard(
+    currentCard: LessonCard,
+    upcomingCards: LessonCard[],
+    completedCards: LessonCard[]
+  ) {
+    if (completedCards.length === numCards) {
       return;
     }
-    const currentCard = lessonCards.current[0];
-    lessonCards.current = lessonCards.current.slice(1);
+    if (currentCard.card.key === activeCard?.key) {
+      return resumeAudio();
+    }
+
+    // TODO: for future lesson types we would update the upcoming cards
+    const updatedCard = await playAudio(currentCard);
+    const next = nextCard(updatedCard, upcomingCards, completedCards);
 
-    await playTermAndDefinition(currentCard, 1);
-    playNextCard();
+    playCard(next.currentCard, next.upcomingCards, next.completedCards);
   }
 
-  function pauseLesson() {
-    pause();
+  function nextCard(
+    currentCard: LessonCard,
+    upcomingCards: LessonCard[],
+    completedCards: LessonCard[]
+  ) {
+    const newcompletedCards = [currentCard, ...completedCards];
+    const newCurrentCard: LessonCard = upcomingCards[0] ?? currentCard;
+    const newUpcomingCards = upcomingCards.length === 0 ? upcomingCards : upcomingCards.slice(1);
```

Comment:
`slice` has a lot of case handling internally so this is equivalent to just `upcomingCards.slice(1)`

from the standard: 
> If start is greater than the index range of the sequence, an empty array is returned.

Reply:
oh dope! it should be fixed now